### DISCUSSION TITLE
IDE Setup/Example project?

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [pmd/pmd#4563] [Q&A]
I would really like to develop my own custom java rules. I found several discussions here, the pmd example projects (https://github.com/pmd/pmd-examples) but i wasn't able to setup a simple project with gradle.

Is there any project with for example gradle, testcases which one can use to start with?

**Comment by jsotuyod** [comment]
No, we don't have such a thing, but you should be able to take the [Maven examples](https://github.com/pmd/pmd-examples/tree/java) and perform an [automatic migration to Gradle](https://docs.gradle.org/current/userguide/migrating_from_maven.html#migmvn:automatic_conversion). I've not tried it, but since the Maven build is extremely simple I would expect it to work pretty well.
